### ISSUE TITLE
Error in compiled Go output

### ISSUE THREAD

**Issue by shayanh** [UBC-NSS/pgo#167]
When I compile the attached spec to Go, the outputted Go program has compile error. The problem is a missing comma somewhere but I couldn't fix it myself.

Attached spec: https://gist.github.com/shayanh/49b5a2a5c4e208e67d0d6fa4ffad84bd

**Comment by fhackett** [comment]
You're right. This one is weird. Still root-causing... can't get any of the tools to say something that makes sense to me, so I'll have to manually bisect this one tomorrow :)

**Comment by fhackett** [comment]
Found it! Turns out, multi-index ops strike again...

I forgot to generate the commas separating indices in the resource read ops.

PR soon.
